lib/tasks.js

```diff
@@ -0,0 +1,397 @@
+'use strict';
+
+const utils = require('./utils');
+const Timer = require('./timer');
+const Task = require('./task');
+const noop = cb => cb();
+
+/**
+ * Static factory method for creating a custom `Composer` class that
+ * extends the given `Emitter`.
+ *
+ * ```js
+ * const Emitter = require('events');
+ * const Composer = require('composer').create(Emitter);
+ * const composer = new Composer();
+ * ```
+ * @name .create
+ * @param {function} `Emitter` Event emitter.
+ * @return {Class} Returns a custom `Composer` class.
+ * @api public
+ */
+
+module.exports = Emitter => {
+
+  /**
+   * Create an instance of `Composer` with the given `options`.
+   *
+   * ```js
+   * const Composer = require('composer');
+   * const composer = new Composer();
+   * ```
+   * @class
+   * @name Composer
+   * @param {object} `options`
+   * @api public
+   */
+
+  class Tasks extends Emitter {
+    constructor(options = {}) {
+      super(options);
+      this.options = options;
+      this.taskStack = new Set();
+      this.taskState = new Map();
```

Comment:
I don't see this used anyplace other than in `resolveBuild`. Is this so tasks have something to add state to?

Reply:
> Is this so tasks have something to add state to?

yes, but I think we should remove it if you haven't already